packages/ui/src/components/dropdown-menu/dropdown-menu.web.tsx

```diff
@@ -0,0 +1,130 @@
+import { ReactNode, forwardRef } from 'react';
+
+import * as RadixDropdownMenu from '@radix-ui/react-dropdown-menu';
+import { css } from 'leather-styles/css';
+import { type HTMLStyledProps, styled } from 'leather-styles/jsx';
+
+import { pressableBaseStyles, pressableStyles } from '../../components/pressable/pressable.web';
+import { ChevronDownIcon } from '../../icons/index';
+import { Flag } from '../flag/flag.web';
+
+export interface DropdownMenuItem {
+  iconLeft?: ReactNode;
+  iconRight?: ReactNode;
+  label: string;
+}
+const dropdownButtonStyles = css({
+  bg: 'ink.background-primary',
+  borderRadius: 'xs',
+  fontWeight: 500,
+  maxWidth: 'fit-content',
+  maxHeight: 'fit-content',
+  px: 'space.04',
+  py: 'space.03',
+  textStyle: 'label.02',
+  userSelect: 'none',
+  '[data-state=open] &': { bg: 'ink.component-background-pressed' },
+});
+
+function Button({ children, ...props }: HTMLStyledProps<'div'>) {
+  return (
+    <styled.div className={dropdownButtonStyles} {...props}>
+      <Flag spacing="space.02" reverse img={<ChevronDownIcon variant="small" />}>
+        {children}
+      </Flag>
+    </styled.div>
+  );
+}
+
+const dropdownIconButtonStyles = css({
+  _hover: { bg: 'ink.component-background-hover' },
+  _focus: { outline: 'none' },
+  p: 'space.02',
+
+  '&[data-state=open]': { bg: 'ink.component-background-pressed' },
+});
+const IconButton: typeof RadixDropdownMenu.Trigger = forwardRef((props, ref) => (
+  <RadixDropdownMenu.Trigger className={dropdownIconButtonStyles} ref={ref} {...props} />
+));
+
+const dropdownTriggerStyles = css({
+  _focus: { outline: 'none' },
+});
+
+const Trigger: typeof RadixDropdownMenu.Trigger = forwardRef((props, ref) => (
+  <RadixDropdownMenu.Trigger className={dropdownTriggerStyles} ref={ref} {...props} />
+));
+
+const dropdownContentStyles = css({
+  alignItems: 'center',
+  '--base-menu-padding': '0px',
+  bg: 'ink.background-primary',
+  borderRadius: 'xs',
+  boxShadow:
+    '0px 12px 24px 0px rgba(18, 16, 15, 0.08), 0px 4px 8px 0px rgba(18, 16, 15, 0.08), 0px 0px 2px 0px rgba(18, 16, 15, 0.08)',
+  p: '0',
+  willChange: 'transform, opacity',
+  zIndex: 999,
+  // TODO re-enable these and check if Settings tests still fail
```

Comment:
This animation is still [causing extension shard 3 to fail](https://github.com/leather-wallet/extension/pull/5518) with timeouts in the `add-network.spec.ts`

Reply:
We have this task https://github.com/leather-wallet/extension/issues/5054 

I should have added it in the comment, I'll add it